requests_async/adapters.py

```diff
@@ -35,7 +35,15 @@ class HTTPAdapter(requests.adapters.HTTPAdapter):
         headers = [("host", urlparts.netloc)] + list(request.headers.items())
 
         conn_kwargs = {'ssl': no_verify()} if urlparts.scheme == 'https' else {}
-        reader, writer = await asyncio.open_connection(hostname, port, **conn_kwargs)
+
+        # if timeout is None, operation waits till complete
+        timeout = kwargs.get('timeout', None)
+        get_conn_coro = self._get_connection(hostname, port, **conn_kwargs)
+
+        try:
+            reader, writer = await asyncio.wait_for(get_conn_coro, timeout)
+        except asyncio.TimeoutError:
+            raise requests.ConnectTimeout()
```

Comment:
Looks good - does requests use the same timeout value both for the connect and for data receiving?

Reply:
By default, it does. But has the option of specifying different timeout values for the connect and read

Reply:
Okay great. I guess we'll want to do something like...

if isinstance(timeout, tuple):
    connect_timeout, read_timeout = timeout
else:
    connect_timeout = timeout
    read_timeout = timeout

Also, we might want to properly expand out the signature of the `HTTPAdapter.send()` method here, instead of using the existing `*args`, `**kwargs`

Reply:
Okay, great. I will fix that up now.

By expanding out, do you mean as it is in requests?
Something like this?
`async def send(self, request, stream=False, timeout=None, verify=True, cert=None, proxies=None)` 
wit the typing of course

Reply:
Yup. Add the type info if requests does, probably don't if it doesn't.